include/boost/url/const_string.hpp

```diff
@@ -45,14 +45,37 @@ class const_string : public string_view
     struct base;
     struct result;
 
-    base* p_;
+    static constexpr
+    std::size_t
+    buffer_size_ = sizeof(base*);
+
+    union data_t {
+        base* p_;
+        char buf_[buffer_size_];
+
+        data_t() = default;
+        data_t(base* p) : p_(p) {}
+    };
+
+    data_t data_;
+
+    static constexpr
+    bool
+    is_small(std::size_t s) noexcept
+    {
+        return s <= buffer_size_;
+    }
 
     BOOST_URL_DECL
     explicit
     const_string(
         result const& r) noexcept;
 
 public:
+    static constexpr
+    std::size_t
+    buffer_size = buffer_size_;
+
```

Comment:
move this to the top and remove `buffer_size_`

Reply:
and bump the size up to 32 :)

Reply:
> and bump the size up to 32 :)

😎